### ISSUE TITLE
Use Flexsearch API instead of hard-coded LIMIT SQL

### ISSUE THREAD

**Issue by marcopelegrini** [Adyen/adyen-hybris#66]
This line: https://github.com/Adyen/adyen-hybris/blob/d246b329a6b6ad210d442160531a7195c24f5342/adyenv6core/src/com/adyen/v6/repository/NotificationItemRepository.java#L42

causes issues because the Flexible search API is not being used, but rather a hardcoded limit sql instruction:

```
de.hybris.platform.servicelayer.search.exceptions.FlexibleSearchException: SQL search error - ORA-00933: SQL command not properly ended
 query = 'SELECT  item_t0.PK  FROM adyennotificationitem item_t0 WHERE ( item_t0.p_processedat  IS NULL) AND (item_t0.TypePkString=? ) order by  item_t0.PK  ASC LIMIT 1000', values = [8796130672722]
	at de.hybris.platform.servicelayer.search.impl.DefaultFlexibleSearchService$2.execute(DefaultFlexibleSearchService.java:416)
	at de.hybris.platform.servicelayer.search.impl.DefaultFlexibleSearchService$2.execute(DefaultFlexibleSearchService.java:1)
	at de.hybris.platform.servicelayer.session.impl.DefaultSessionService.executeInLocalView(DefaultSessionService.java:89)
	at de.hybris.platform.servicelayer.search.impl.DefaultFlexibleSearchService.getJaloResult(DefaultFlexibleSearchService.java:395)
	at de.hybris.platform.servicelayer.search.impl.DefaultFlexibleSearchService.search(DefaultFlexibleSearchService.java:167)
	at com.adyen.v6.repository.NotificationItemRepository.getNonProcessedNotifications(NotificationItemRepository.java:46)
	at com.adyen.v6.cronjob.AdyenProcessNotificationCronJob.perform(AdyenProcessNotificationCronJob.java:50)
	at de.hybris.platform.servicelayer.internal.jalo.ServicelayerJob.performCronJob(ServicelayerJob.java:38)
	at de.hybris.platform.cronjob.jalo.Job.execute(Job.java:1358)
	at de.hybris.platform.cronjob.jalo.Job.performImpl(Job.java:814)
	at de.hybris.platform.cronjob.jalo.Job.access$1(Job.java:767)
	at de.hybris.platform.cronjob.jalo.Job$JobRunable.run(Job.java:686)
	at de.hybris.platform.util.threadpool.PoolableThread.run(PoolableThread.java:195)
```

**Comment by rikterbeek** [comment]
Hi @marcopelegrini,

We created a custom query used inside the flexibleSearchService.  What better approach would you suggest ?

Regards,
Rik
Adyen

**Comment by marcopelegrini** [comment]
You just need to use `FlexibleSearchQuery#setCount` instead of setting the limit hard-coded in the query, the platform will take care of adjusting the query according to the database.

**Comment by lancergr** [comment]
Fixed in #67
